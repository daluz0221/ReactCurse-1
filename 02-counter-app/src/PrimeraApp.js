import React from 'react'
// import React, {Fragment} from 'react'
import PropTypes from 'prop-types';


const PrimeraApp = ( {saludos, subtitle='soy un fan tuyo'} ) => {
    
 

   
    
    
    return (
        <>
        <h1>{ saludos }</h1>
        {/* <h1>{ JSON.stringify(saludo, null, 3) }</h1> */}
        <p>{ subtitle }</p>
        </>
    );
    
};

PrimeraApp.propTypes = {
    saludos: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle: 'soy fan de mimosa'
}

export default PrimeraApp;
