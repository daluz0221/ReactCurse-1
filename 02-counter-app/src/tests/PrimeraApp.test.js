import React from 'react'
import '@testing-library/jest-dom';
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme'

describe('Pruebas en componente <PrimeraApp>', () => {
    
    
    // test('debe mostrar el mensaje "Hola noelle"', () => {
       
    //     const saludo = 'Hola noelle';

    //     const {getByText} = render( <PrimeraApp saludos={saludo}/> )

    //     expect(getByText(saludo)).toBeInTheDocument();

    // });
    

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        
        const saludo = 'Hola noelle';
        const wrapper = shallow(<PrimeraApp saludos={saludo}/>);

        expect( wrapper ).toMatchSnapshot();

    });
    

    test('debe de mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola noelle';
        const subtitulo = 'Quiero ser tu amigo';
        const wrapper = shallow(
        <PrimeraApp 
            saludos={saludo}
            subtitle={subtitulo}
            
        />);
        // con este find busco la etiqueta u objeto dentro del html del componente
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitulo)

    });
    

});
