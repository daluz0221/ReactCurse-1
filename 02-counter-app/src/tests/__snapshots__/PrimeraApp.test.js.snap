// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <PrimeraApp> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola noelle
  </h1>
  <p>
    soy tu fan
  </p>
</Fragment>
`;
